src/SprykerMiddleware/Zed/Process/Business/Aggregator/BatchAggregator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace SprykerMiddleware\Zed\Process\Business\Aggregator;
+
+use Generated\Shared\Transfer\AggregatorSettingsTransfer;
+use SprykerMiddleware\Zed\Process\Business\Renderer\RendererInterface;
+use SprykerMiddleware\Zed\Process\Business\Writer\WriterInterface;
+
+class BatchAggregator implements AggregatorInterface
+{
+    /**
+     * @var \SprykerMiddleware\Zed\Process\Business\Writer\WriterInterface
+     */
+    protected $writer;
+
+    /**
+     * @var \SprykerMiddleware\Zed\Process\Business\Renderer\RendererInterface
+     */
+    protected $renderer;
+
+    /**
+     * @var \Generated\Shared\Transfer\AggregatorSettingsTransfer|\Generated\Shared\Transfer\IteratorSettingsTransfer|null
+     */
+    protected $settings;
+
+    /**
+     * @var array
+     */
+    protected $storage = [];
+
+    /**
+     * BatchAggregator constructor.
+     *
+     * @param \SprykerMiddleware\Zed\Process\Business\Writer\WriterInterface $writer
+     * @param \SprykerMiddleware\Zed\Process\Business\Renderer\RendererInterface $renderer
+     * @param \Generated\Shared\Transfer\AggregatorSettingsTransfer $settings
+     */
+    public function __construct(WriterInterface $writer, RendererInterface $renderer, AggregatorSettingsTransfer $settings)
+    {
+        $this->writer = $writer;
+        $this->renderer = $renderer;
```

Comment:
we have to have additional discussion on how the renderer is supposed to be wired up, because imho it is not needed in most writer context and should thus be directly configured into the writer if needed.